protocol/get_ledgers.go

```diff
@@ -33,21 +27,10 @@ type GetLedgersRequest struct {
 
 // validate checks the validity of the request parameters.
 func (req *GetLedgersRequest) Validate(maxLimit uint, ledgerRange LedgerSeqRange) error {
-	switch {
-	case req.Pagination != nil:
-		switch {
-		case req.Pagination.Cursor != "" && req.StartLedger != 0:
-			return errors.New("startLedger and cursor cannot both be set")
-		case req.Pagination.Limit > maxLimit:
-			return fmt.Errorf("limit must not exceed %d", maxLimit)
-		}
-	case req.StartLedger != 0 && !IsStartLedgerWithinBounds(req.StartLedger, ledgerRange):
-		return fmt.Errorf(
-			"start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance",
-			ledgerRange.FirstLedger,
-			ledgerRange.LastLedger,
-		)
-	}
+	return errors.Join(
+		ValidatePagination(req.StartLedger, req.Pagination, maxLimit, ledgerRange),
+		IsValidFormat(req.Format),
+	) // nils will coalesce
```

Comment:
Thanks for adding the nils comment. The semantics of the `error` APIs are rarely obvious to me.

Reply:
Yep, a little too subtle for my taste but it does make this a clean call!